contrib_bots/lib/tictactoe-bot.py

```diff
@@ -0,0 +1,345 @@
+
+from __future__ import absolute_import
+from __future__ import print_function
+import copy
+import random
+from six.moves import range
+
+initial_board = [["_", "_", "_"],
+                 ["_", "_", "_"],
+                 ["_", "_", "_"]]
+
+mode = 'r'  # default, can change for debugging to 'p'
+def output_mode(string_to_print, mode):
+    if mode == "p":
+        print(string_to_print)
+    elif mode == "r":
+        return string_to_print
+
+# -------------------------------------
+class TicTacToeGame(object):
+    smarter = True
+    # If smarter is True, the computer will do some extra thinking when it makes its move - it'll be harder for the user.
+
+    triplets = [[(0, 0), (0, 1), (0, 2)],  # Row 1
+                [(1, 0), (1, 1), (1, 2)],  # Row 2
+                [(2, 0), (2, 1), (2, 2)],  # Row 3
+                [(0, 0), (1, 0), (2, 0)],  # Column 1
+                [(0, 1), (1, 1), (2, 1)],  # Column 2
+                [(0, 2), (1, 2), (2, 2)],  # Column 3
+                [(0, 0), (1, 1), (2, 2)],  # Diagonal 1
+                [(0, 2), (1, 1), (2, 0)]   # Diagonal 2
+                ]
+
+    positions = "Here are the coordinates of each position. (Parentheses and spaces are optional.) \n" \
+                "(1, 1)  (1, 2)  (1, 3) \n(2, 1)  (2, 2)  (2, 3) \n(3, 1) (3, 2) (3, 3) \n " \
```

Comment:
I guess you need to specify which item within a coordinate represents a column or row.